storage/store.go

```diff
@@ -568,7 +568,7 @@ func (s *Store) startGossip() {
 	ctx := s.Context(nil)
 	// Periodic updates run in a goroutine and signal a WaitGroup upon completion
 	// of their first iteration.
-	s.initComplete.Add(1)
+	s.initComplete.Add(2)
```

Comment:
I never observed this actually racing, so you're probably right. I noticed
it while chasing the feed bug and fixed it on instinct.
On Jul 27, 2015 8:14 PM, "Ben Darnell" <notifications@github.com> wrote:

> In storage/store.go
> <https://github.com/cockroachdb/cockroach/pull/1827#discussion_r35602927>:
>
> > @@ -568,7 +568,7 @@ func (s *Store) startGossip() {
> >  	ctx := s.Context(nil)
> >  	// Periodic updates run in a goroutine and signal a WaitGroup upon completion
> >  	// of their first iteration.
> > -	s.initComplete.Add(1)
> > +	s.initComplete.Add(2)
>
> I'm surprised this was a race. Don't both calls to Add() definitely
> happen before any call to WaitForInit?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/1827/files#r35602927>.
>